Make local_binaries respect bindings in parent

rgrinberg opened the pull request:
Previously, those bindings were ignored

rgrinberg committed:
Make local_binaries respect bindings in parent

Previously, those bindings were ignored

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

aalekseyev commented:
This sounds like something that could benefit from a test demonstrating the feature.

rgrinberg committed:
Add locations to local bindings

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg committed:
Re-org File_binding.t module

Make File_binding.t the singular form and put the previous bindings list under
File_binding.L.t

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg committed:
Create dedicated types for Expanded/Unexpanded file bindings

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg committed:
Fix the type of File_bindings.Expanded.t

In fact, the path of the source is decided at expansion time. So the source
should be Path.t rather than string

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg committed:
Add test for nested local binaries

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg commented:
Good call on suggesting a test. There was actually another serious bug present with regards to expansion. The path of the source directory was decided relative to where the node was requested. This is wrong and the path should just be set at expansion time. I've added `File_binding.{Unexpanded,Expanded}.t` to make this all clearer.

I've also added some improvements to the location handling. This should improve error messages.

rgrinberg committed:
Update CHANGES

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>
